packages/compose/src/hooks/use-fresh-ref/index.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef } from '@wordpress/element';
+
+/**
+ * A ref that always contains the lastest value.
+ *
+ * @param {any} value The value to store in the ref.
+ *
+ * @return {import('react').RefObject} A ref object with the value.
+ */
+export default function useFreshRef( value ) {
```

Comment:
I believe a more common name would be `useLatest` or `useLatestRef`, as also seen in [`react-use`](https://github.com/streamich/react-use/blob/master/docs/useLatest.md)

Reply:
I like `useLatestRef`. Sounds good! 🙂

Reply:
I think we should also update the PR title and description 😉.